tests/ModelContextProtocol.Tests/Configuration/McpServerBuilderExtensionsToolsTests.cs

```diff
@@ -765,4 +767,16 @@ public class ObjectWithId
     {
         public string Id { get; set; } = Guid.NewGuid().ToString("N");
     }
+
+    [JsonSerializable(typeof(bool))]
+    [JsonSerializable(typeof(int))]
+    [JsonSerializable(typeof(long))]
+    [JsonSerializable(typeof(double))]
+    [JsonSerializable(typeof(string))]
```

Comment:
Not directly related to this PR, but seems like it'd be a helpful thing if we imbued AIJsonUtilities.DefaultOptions with source generation for all the primitive types, if it doesn't have that already. It'd help to make more things "just work".